packages/lodestar-validator/src/slashingProtection/attestation/dbRepository.ts

```diff
@@ -0,0 +1,71 @@
+import {
+  BLSPubkey,
+  Epoch,
+  SlashingProtectionAttestation,
+  SlashingProtectionAttestationLowerBound,
+} from "@chainsafe/lodestar-types";
+import {intToBytes} from "@chainsafe/lodestar-utils";
+import {IDatabaseController, Bucket, encodeKey, IDatabaseApiOptions} from "@chainsafe/lodestar-db";
+import {Type} from "@chainsafe/ssz";
+
+export class AttestationByTargetRepository {
+  protected type: Type<SlashingProtectionAttestation>;
+  protected db: IDatabaseController<Buffer, Buffer>;
+  protected bucket = Bucket.slashingProtectionAttestationByTarget;
+
+  constructor(opts: IDatabaseApiOptions) {
+    this.db = opts.controller;
+    this.type = opts.config.types.SlashingProtectionAttestation;
+  }
+
+  async getAll(pubkey: BLSPubkey, limit?: number): Promise<SlashingProtectionAttestation[]> {
+    const blocks = await this.db.values({
+      limit,
+      gte: this.encodeKey(pubkey, 0),
+      lt: this.encodeKey(pubkey, Number.MAX_SAFE_INTEGER),
+    });
+    return blocks.map((block) => this.type.deserialize(block));
+  }
+
+  async get(pubkey: BLSPubkey, targetEpoch: Epoch): Promise<SlashingProtectionAttestation | null> {
+    const att = await this.db.get(this.encodeKey(pubkey, targetEpoch));
+    return att && this.type.deserialize(att);
+  }
+
+  async set(pubkey: BLSPubkey, atts: SlashingProtectionAttestation[]): Promise<void> {
+    await this.db.batchPut(
+      atts.map((att) => ({
+        key: this.encodeKey(pubkey, att.targetEpoch),
+        value: Buffer.from(this.type.serialize(att)),
+      }))
+    );
+  }
+
+  private encodeKey(pubkey: BLSPubkey, targetEpoch: Epoch): Buffer {
+    return encodeKey(this.bucket, Buffer.concat([Buffer.from(pubkey), intToBytes(BigInt(targetEpoch), 8, "be")]));
+  }
+}
+
+export class AttestationLowerBoundRepository {
```

Comment:
can we keep each class in separate file?

I don't think it's just me since I noticed it on other colleagues, when you open a file if first class name you see isn't what you looking for you will go to next file

Reply:
Yes I agree, I'll split it in a new PR

Reply:
New PR https://github.com/ChainSafe/lodestar/pull/1679